.github/workflows/document-syntax.yml

```diff
@@ -0,0 +1,35 @@
+name: Syntax changes reminder
+
+on:
+  pull_request:
+    paths:
+      - 'ocaml/parsing/parser.mly'
+
+jobs:
+  remind:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Create PR Comment
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          REPO=${{ github.repository }}
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          COMMENT="## Parser Change Checklist
+
+          This PR modifies the parser. Please check that the following tests are updated:
+
+          - [ ] \`parsetree/source_jane_street.ml\`
+          - [ ] \`language-extensions/pprintast_unconditional.ml\`
```

Comment:
It would be wonderful to delete `language-extensions/pprintast_unconditional.ml`. I think I will do this in a follow-on PR after this one is merged.